### ISSUE TITLE
LXT on vit-like models

### ISSUE THREAD

**Issue by Lotto98** [rachtibat/LRP-eXplains-Transformers#31]
Hi,

thank you for this implementation!

I'm trying to apply LRP to a batch of images, is this code correct? 

```python
input_tensor.requires_grad_().retain_grad()  # Retain gradients for input tensor: this is necessary for not leaf tensors

# For each sample in batch, compute gradient of output w.r.t input
grads = []
for i in range(input_tensor.shape[0]):
    self.model.zero_grad()
    
    output = self.model(input_tensor[i].unsqueeze(0))  # (1, 3, 224, 224)
    
    output[0,target_classes[i]].backward(retain_graph=True)
    
    grads.append(input_tensor.grad[i].clone())

# Convert grads to tensor
grads = torch.stack(grads)

attributions = (grads * input_tensor)
```

I'm working with two different vit-like models:
- deit_base_patch16_224 from Timm library
- MedViT: https://github.com/Omid-Nejati/MedViT

Deit model is monkey patched with vit_torch.cp_LRP, while for MedViT I created this function to detach q and k since it uses not standard functions:

```python
def E_MHSA_no_grad(self, x):
    B, N, C = x.shape
    q = self.q(x)
    q = q.reshape(B, N, self.num_heads, int(C // self.num_heads)).permute(0, 2, 1, 3)

    if self.sr_ratio > 1:
        x_ = x.transpose(1, 2)
        x_ = self.sr(x_)
        if not torch.onnx.is_in_onnx_export() and not self.is_bn_merged:
            x_ = self.norm(x_)
        x_ = x_.transpose(1, 2)
        k = self.k(x_)
        k = k.reshape(B, -1, self.num_heads, int(C // self.num_heads)).permute(0, 2, 3, 1)
        v = self.v(x_)
        v = v.reshape(B, -1, self.num_heads, int(C // self.num_heads)).permute(0, 2, 1, 3)
    else:
        k = self.k(x)
        k = k.reshape(B, -1, self.num_heads, int(C // self.num_heads)).permute(0, 2, 3, 1)
        v = self.v(x)
        v = v.reshape(B, -1, self.num_heads, int(C // self.num_heads)).permute(0, 2, 1, 3)
    
    # Detach the tensors to avoid gradient computation
    q = q.detach()
    k = k.detach()
    
    #print(q, flush=True)
    #print(k, flush=True)
    
    print("PATCHED")

    attn = (q @ k) * self.scale

    attn = attn.softmax(dim=-1)
    attn = self.attn_drop(attn)

    x = (attn @ v).transpose(1, 2).reshape(B, N, C)
    x = self.proj(x)
    x = self.proj_drop(x)
    return x
```

I would really appreciate your opinion on what i have done, also if you have some other tips I would love to hear them! 

Thank you
